src/screens/Testing/TestProcess/index.tsx

```diff
@@ -11,162 +11,124 @@ import {ProgressType} from '@src/components/ProgressBar/ProgressView';
+            [...currentTest[0].content.correctAnswer].sort().join('').toLowerCase()
+        ) {
+          return 'error';
+        } else if (checkedAnswer.length === 0) {
+          return 'default';
+        }
+        break;
       }
+      default:
+        return 'active';
     }
   };
+  const setNextResult = (
+    questionStatus: 'default' | 'active' | 'right' | 'error' | undefined,
+  ) => {
+    dispatch(
+      setStateResult({
+        id: currentTest[0].id,
+        questionStatus,
+        answer: currentTest[0].type === 'single' ? singleAnswer.join() : checkedAnswer.join(),
+      }),
+    );
+  };
   const onPressNextAnswer = () => {
     if (
-      numAnswer < quizIdMocState.questions.length &&
-      resultData.length < quizIdMocState.questions.length
+      numAnswer < quizIdData.questions.length &&
+      resultData.length < quizIdData.questions.length
     ) {
-      progressResult();
+      setNextResult(progressResult());
       setNumAnswer(numAnswer + 1);
       setSingleAnswer([]);
       setIsActiveRadio(undefined);
     }
     if (
-      numAnswer === quizIdMocState.questions.length &&
-      resultData.length < quizIdMocState.questions.length
+      numAnswer === quizIdData.questions.length &&
+      resultData.length < quizIdData.questions.length
     ) {
       navigate(ScreenList.TESTS, {screen: ScreenList.TEST_RESULT});
-      progressResult();
+      setNextResult(progressResult());
       setSingleAnswer([]);
     }
   };
+  const setSkipResult = () => {
+    dispatch(
+      setStateResult({
+        id: currentTest[0].id,
+        questionStatus: 'default',
+        answer: currentTest[0].type === 'single' ? singleAnswer.join() : checkedAnswer.join(),
+      }),
+    );
+  };
   const onPressSkipAnswer = () => {
-    if (numAnswer < quizIdMocState.questions.length) {
+    if (numAnswer < quizIdData.questions.length) {
       setNumAnswer(numAnswer + 1);
       setSingleAnswer([]);
       if (currentTest[0].type === 'single') {
-        dispatch(
-          setStateResult({
-            id: currentTest[0].id,
-            questionStatus: 'default',
-            answer: singleAnswer.join(),
-          }),
-        );
+        setSkipResult();
+      } else if (currentTest[0].type === 'multi') {
+        setSkipResult();
       }
-      if (currentTest[0].type === 'multi') {
-        dispatch(
-          setStateResult({
-            id: currentTest[0].id,
-            questionStatus: 'default',
-            answer: checkAnswer.join(),
-          }),
-        );
-      }
-    }
-
-    if (numAnswer === quizIdMocState.questions.length) {
+    } else if (numAnswer === quizIdData.questions.length) {
       navigate(ScreenList.TESTS, {screen: ScreenList.TEST_RESULT});
       setSingleAnswer([]);
-
-      dispatch(
-        setStateResult({
-          id: currentTest[0].id,
-          questionStatus: 'default',
-          answer: singleAnswer.join(),
-        }),
-      );
+      setSkipResult();
```

Comment:
you can invoke `setSkipResult ` after all conditions because you invoke it in all cases